engine/src/main/java/io/zeebe/engine/processing/streamprocessor/ReProcessingStateMachine.java

```diff
@@ -344,6 +372,12 @@ private void reprocessRecord(final TypedRecord<?> currentEvent) {
 
         eventApplier.applyState(
             currentEvent.getKey(), currentEvent.getIntent(), currentEvent.getValue());
+
+      } else if (currentEvent.getRecordType() == RecordType.COMMAND) {
+        // restore the key generator because it is used by not yet migrated processors
+        Optional.ofNullable(followUpKeyBySourceCommandPosition.get(currentEvent.getPosition()))
+            .filter(key -> key > keyGeneratorControls.getCurrentKey())
+            .ifPresent(keyGeneratorControls::setKey);
```

Comment:
You could also expose setKeyIfHigher(key) in KeyGeneratorControls. Then you don't have to expose the `getCurrentKey` as a method, but allow the Controls class to add this Max.(x,y) logic